### ISSUE TITLE
E tensorflow/stream_executor/rocm/rocm_driver.cc:600] failed to allocate 890.06M (933298432 bytes) from device: HIP_ERROR_OutOfMemory

### ISSUE THREAD

**Issue by sumannelli** [keras-team/keras#13889]
while running the cnn models in jupyter notebook getting the erros as:
 
E tensorflow/stream_executor/rocm/rocm_driver.cc:600] failed to allocate 890.06M (933298432 bytes) from device: HIP_ERROR_OutOfMemory
Downloaded the jupyter notebook from the below link, when training the model getting the error

https://colab.research.google.com/drive/1ldnii3sGJaUHPV6TWImykbeE_O-8VIIN

- OS Platform and Distribution: Ubuntu 18.04
- TensorFlow backend : yes
- TensorFlow version:  2.1.0
- Keras version:  2.3.1
- Python version:  3.6.9
- CUDA/cuDNN version:  
- GPU model and memory:  AMD radeon vii (vega 20)
processor : Intel® Core™ i7-3770 CPU @ 3.40GHz × 8
Ram : 32 GB
kernel:5

Using TensorFlow backend.
Creating model, this may take a second...
2020-03-11 17:17:25.048625: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libhip_hcc.so
2020-03-11 17:17:25.109305: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1573] Found device 0 with properties: 
pciBusID: 0000:03:00.0 name: Vega 20     ROCm AMD GPU ISA: gfx906
coreClock: 1.801GHz coreCount: 60 deviceMemorySize: 15.98GiB deviceMemoryBandwidth: -1B/s
2020-03-11 17:17:25.146781: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library librocblas.so
2020-03-11 17:17:25.148464: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libMIOpen.so
2020-03-11 17:17:25.149749: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library librocfft.so
2020-03-11 17:17:25.149973: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library librocrand.so
2020-03-11 17:17:25.150054: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1697] Adding visible gpu devices: 0
2020-03-11 17:17:25.150389: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE3 SSE4.1 SSE4.2 AVX
2020-03-11 17:17:25.155184: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 3403420000 Hz
2020-03-11 17:17:25.155518: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x80d58d0 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2020-03-11 17:17:25.155541: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
2020-03-11 17:17:25.156543: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1573] Found device 0 with properties: 
pciBusID: 0000:03:00.0 name: Vega 20     ROCm AMD GPU ISA: gfx906
coreClock: 1.801GHz coreCount: 60 deviceMemorySize: 15.98GiB deviceMemoryBandwidth: -1B/s
2020-03-11 17:17:25.156590: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library librocblas.so
2020-03-11 17:17:25.156627: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libMIOpen.so
2020-03-11 17:17:25.156643: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library librocfft.so
2020-03-11 17:17:25.156659: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library librocrand.so
2020-03-11 17:17:25.156700: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1697] Adding visible gpu devices: 0
2020-03-11 17:17:25.156746: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1096] Device interconnect StreamExecutor with strength 1 edge matrix:
2020-03-11 17:17:25.156757: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1102]      0 
2020-03-11 17:17:25.156762: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1115] 0:   N 
2020-03-11 17:17:25.156849: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1241] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 15306 MB memory) -> physical GPU (device: 0, name: Vega 20, pci bus id: 0000:03:00.0)
2020-03-11 17:17:25.172238: E tensorflow/stream_executor/rocm/rocm_driver.cc:600] failed to allocate 14.95G (16049923584 bytes) from device: HIP_ERROR_OutOfMemory
2020-03-11 17:17:25.172299: E tensorflow/stream_executor/rocm/rocm_driver.cc:600] failed to allocate 13.45G (14444931072 bytes) from device: HIP_ERROR_OutOfMemory
2020-03-11 17:17:25.172334: E tensorflow/stream_executor/rocm/rocm_driver.cc:600] failed to allocate 12.11G (13000437760 bytes) from device: HIP_ERROR_OutOfMemory
2020-03-11 17:17:25.172365: E tensorflow/stream_executor/rocm/rocm_driver.cc:600] failed to allocate 10.90G (11700393984 bytes) from device: HIP_ERROR_OutOfMemory
2020-03-11 17:17:25.172396: E tensorflow/stream_executor/rocm/rocm_driver.cc:600] failed to allocate 9.81G (10530354176 bytes) from device: HIP_ERROR_OutOfMemory
2020-03-11 17:17:25.172427: E tensorflow/stream_executor/rocm/rocm_driver.cc:600] failed to allocate 8.83G (9477318656 bytes) from device: HIP_ERROR_OutOfMemory
2020-03-11 17:17:25.172457: E tensorflow/stream_executor/rocm/rocm_driver.cc:600] failed to allocate 7.94G (8529586688 bytes) from device: HIP_ERROR_OutOfMemory
2020-03-11 17:17:25.172488: E tensorflow/stream_executor/rocm/rocm_driver.cc:600] failed to allocate 7.15G (7676627968 bytes) from device: HIP_ERROR_OutOfMemory
2020-03-11 17:17:25.172518: E tensorflow/stream_executor/rocm/rocm_driver.cc:600] failed to allocate 6.43G (6908964864 bytes) from device: HIP_ERROR_OutOfMemory
2020-03-11 17:17:25.172548: E tensorflow/stream_executor/rocm/rocm_driver.cc:600] failed to allocate 5.79G (6218067968 bytes) from device: HIP_ERROR_OutOfMemory
2020-03-11 17:17:25.172578: E tensorflow/stream_executor/rocm/rocm_driver.cc:600] failed to allocate 5.21G (5596260864 bytes) from device: HIP_ERROR_OutOfMemory
2020-03-11 17:17:25.172608: E tensorflow/stream_executor/rocm/rocm_driver.cc:600] failed to allocate 4.69G (5036634624 bytes) from device: HIP_ERROR_OutOfMemory
2020-03-11 17:17:25.172637: E tensorflow/stream_executor/rocm/rocm_driver.cc:600] failed to allocate 4.22G (4532971008 bytes) from device: HIP_ERROR_OutOfMemory
2020-03-11 17:17:25.172667: E tensorflow/stream_executor/rocm/rocm_driver.cc:600] failed to allocate 3.80G (4079673856 bytes) from device: HIP_ERROR_OutOfMemory
2020-03-11 17:17:25.172697: E tensorflow/stream_executor/rocm/rocm_driver.cc:600] failed to allocate 3.42G (3671706368 bytes) from device: HIP_ERROR_OutOfMemory
2020-03-11 17:17:25.172727: E tensorflow/stream_executor/rocm/rocm_driver.cc:600] failed to allocate 3.08G (3304535552 bytes) from device: HIP_ERROR_OutOfMemory
2020-03-11 17:17:25.172756: E tensorflow/stream_executor/rocm/rocm_driver.cc:600] failed to allocate 2.77G (2974081792 bytes) from device: HIP_ERROR_OutOfMemory
2020-03-11 17:17:25.172786: E tensorflow/stream_executor/rocm/rocm_driver.cc:600] failed to allocate 2.49G (2676673536 bytes) from device: HIP_ERROR_OutOfMemory
2020-03-11 17:17:25.172816: E tensorflow/stream_executor/rocm/rocm_driver.cc:600] failed to allocate 2.24G (2409006080 bytes) from device: HIP_ERROR_OutOfMemory
2020-03-11 17:17:25.172845: E tensorflow/stream_executor/rocm/rocm_driver.cc:600] failed to allocate 2.02G (2168105472 bytes) from device: HIP_ERROR_OutOfMemory
2020-03-11 17:17:25.172875: E tensorflow/stream_executor/rocm/rocm_driver.cc:600] failed to allocate 1.82G (1951294976 bytes) from device: HIP_ERROR_OutOfMemory
2020-03-11 17:17:25.172905: E tensorflow/stream_executor/rocm/rocm_driver.cc:600] failed to allocate 1.63G (1756165376 bytes) from device: HIP_ERROR_OutOfMemory
2020-03-11 17:17:25.172935: E tensorflow/stream_executor/rocm/rocm_driver.cc:600] failed to allocate 1.47G (1580548864 bytes) from device: HIP_ERROR_OutOfMemory
2020-03-11 17:17:25.172965: E tensorflow/stream_executor/rocm/rocm_driver.cc:600] failed to allocate 1.32G (1422493952 bytes) from device: HIP_ERROR_OutOfMemory
2020-03-11 17:17:25.172994: E tensorflow/stream_executor/rocm/rocm_driver.cc:600] failed to allocate 1.19G (1280244480 bytes) from device: HIP_ERROR_OutOfMemory
2020-03-11 17:17:25.173024: E tensorflow/stream_executor/rocm/rocm_driver.cc:600] failed to allocate 1.07G (1152220160 bytes) from device: HIP_ERROR_OutOfMemory
2020-03-11 17:17:25.173053: E tensorflow/stream_executor/rocm/rocm_driver.cc:600] failed to allocate 988.96M (1036998144 bytes) from device: HIP_ERROR_OutOfMemory
2020-03-11 17:17:25.173083: E tensorflow/stream_executor/rocm/rocm_driver.cc:600] failed to allocate 890.06M (933298432 bytes) from device: HIP_ERROR_OutOfMemory

Thanks 
Suman Nelli

**Comment by eduardofv** [comment]
The colab looks fine, couldn't find the error. Is your issue solved?
Anyways, I've seen this error when training batches are too big, lowering them usually works although you may need to change the learning rate as well.

**Comment by sumannelli** [comment]
Thanks for the reply. @eduardofv 
No still the issue is persist...I used batch size as 10 and the image size is 660/660..
When looked into rocm-smi, showing the GPU vram is 100% used during the training. When I stopping the training or after completion of the training, the memory is showing as 100% used. I think there is memory lock. I am using this in Jupiter notebook

**Comment by eduardofv** [comment]
What about trying a batch size of 1, just to be sure if it's a memory issue on the GPU?

**Comment by eduardofv** [comment]
Mmmm... I missed that it is a AMD GPU. I don't know if it's compatible with NVIDIA or something but only NVIDIA cards are supported by Tensorflow. 

Either if it's or not the problem I may also recommend running it CPU only to check if the problem is with the GPU or somewhere else.

**Comment by sumannelli-Ib** [comment]
Yes, Amd vega 20 GPU is not working with TensorFlow 1.x, but it's working with Tensorflow 2.1.0. It starting the training process and after completion of the training, the memory is locking showing as VRam 100% used.  When I am again running another model, I am getting the above error.
To avoid this problem, I am restarting my system and memory Vram is releasing.

**Comment by eduardofv** [comment]
I see! I read some time ago about ROCm but dismissed it as it looked far away. It's good to know it works now. May I suggest to post this issue on ROCm forums as it looks more of an undelying problem with the platform rather than TF or Keras? You may get more help from them!
Best, EF

**Comment by sumannelli-Ib** [comment]
Thank you @eduardofv

**Comment by sushreebarsa** [comment]
@sumannelli Moving this issue to closed status as there has been no recent activity, in case you still face the error please create a new issue.Thanks!

**Comment by dipietrantonio** [comment]
Hi All,

I get the same error using ROCm 5.6 and TensorFlow 2.12 (tensorflow-rocm).

```
cdipietrantonio@nid002070:/scratch/pawsey0001/cdipietrantonio/models> python3 01_horovod_mnist.py 
2023-11-01 16:19:17.812101: E tensorflow/compiler/xla/stream_executor/rocm/rocm_driver.cc:606] failed to allocate 127.99TiB (140729850295296 bytes) from device: HIP_ERROR_OutOfMemory
2023-11-01 16:19:17.812133: E tensorflow/compiler/xla/stream_executor/rocm/rocm_driver.cc:606] failed to allocate 115.19TiB (126656857505792 bytes) from device: HIP_ERROR_OutOfMemory
2023-11-01 16:19:17.812142: E tensorflow/compiler/xla/stream_executor/rocm/rocm_driver.cc:606] failed to allocate 103.67TiB (113991166722048 bytes) from device: HIP_ERROR_OutOfMemory
2023-11-01 16:19:17.812150: E tensorflow/compiler/xla/stream_executor/rocm/rocm_driver.cc:606] failed to allocate 93.31TiB (102592046694400 bytes) from device: HIP_ERROR_OutOfMemory
2023-11-01 16:19:17.812157: E tensorflow/compiler/xla/stream_executor/rocm/rocm_driver.cc:606] failed to allocate 83.98TiB (92332837830656 bytes) from device: HIP_ERROR_OutOfMemory
2023-11-01 16:19:17.812163: E tensorflow/compiler/xla/stream_executor/rocm/rocm_driver.cc:606] failed to allocate 75.58TiB (83099555725312 bytes) from device: HIP_ERROR_OutOfMemory
2023-11-01 16:19:17.812169: E tensorflow/compiler/xla/stream_executor/rocm/rocm_driver.cc:606] failed to allocate 68.02TiB (74789599313920 bytes) from device: HIP_ERROR_OutOfMemory
2023-11-01 16:19:17.812176: E tensorflow/compiler/xla/stream_executor/rocm/rocm_driver.cc:606] failed to allocate 61.22TiB (67310639382528 bytes) from device: HIP_ERROR_OutOfMemory
2023-11-01 16:19:17.812182: E tensorflow/compiler/xla/stream_executor/rocm/rocm_driver.cc:606] failed to allocate 55.10TiB (60579574185984 bytes) from device: HIP_ERROR_OutOfMemory
2023-11-01 16:19:17.812188: E tensorflow/compiler/xla/stream_executor/rocm/rocm_driver.cc:606] failed to allocate 49.59TiB (54521615089664 bytes) from device: HIP_ERROR_OutOfMemory
2023-11-01 16:19:17.812194: E tensorflow/compiler/xla/stream_executor/rocm/rocm_driver.cc:606] failed to allocate 44.63TiB (49069451902976 bytes) from device: HIP_ERROR_OutOfMemory
2023-11-01 16:19:17.812200: E tensorflow/compiler/xla/stream_executor/rocm/rocm_driver.cc:606] failed to allocate 40.17TiB (44162506293248 bytes) from device: HIP_ERROR_OutOfMemory
2023-11-01 16:19:17.812206: E tensorflow/compiler/xla/stream_executor/rocm/rocm_driver.cc:606] failed to allocate 36.15TiB (39746256502784 bytes) from device: HIP_ERROR_OutOfMemory
2023-11-01 16:19:17.812212: E tensorflow/compiler/xla/stream_executor/rocm/rocm_driver.cc:606] failed to allocate 32.53TiB (35771629174784 bytes) from device: HIP_ERROR_OutOfMemory
2023-11-01 16:19:17.812218: E tensorflow/compiler/xla/stream_executor/rocm/rocm_driver.cc:606] failed to allocate 29.28TiB (32194464579584 bytes) from device: HIP_ERROR_OutOfMemor
```
